### PR TITLE
Fix `OmegaConfigLoader` __repr__ and missing keys method

### LINKED ISSUE #3618: OmegaConfigLoader's `__repr__` is not implemented correctly, and it does not behave similar to a `UserDict`
## Description

Currently, `base_env` and `default_run_env` is not included in the `__repr__`.

When it is printed, it shows something like this

> In [11]: c
> Out[11]: OmegaConfigLoader(conf_source=/Users/Nok_Lam_Chan/dev/kedro-inspect/demo-project/conf, env=None, config_patterns={'catalog': ['catalog*', 'catalog*/**', '**/catalog*'], 'parameters': ['parameters*', 'parameters*/**', '**/parameters*'], 'credentials': ['credentials*', 'credentials*/**', '**/credentials*'], 'globals': ['globals.yml']})
> 

## Context

Even it inherits from `UserDict`, but the dictionary interface seems broken. For example:
```
for key in config_loader.keys():
  print(key)
```
will print nothing even though `config_loader["parameters"]` exist. For the context, I was trying to build a plugin that iterate the config loader and "compile" the configuration.

I need to use a workaround looks like this:
```python
keys = config_loader.config_patterns.keys()
for key in keys:
   config = config_loader[key] 
   ...
```

 In addition, if `__repr__` is removed, the config_loader will be printed as if it is a "dictionary". Right now it always print the `__repr__` regardless of the `keys`, `values`, `items` or `print` method.

Cc @merelcht

### PR DESCRIPTION
**Pull request by noklam**
## Description

Fix #3618, since we are trying to improve the interactive experience. While it's still a bit unclear as `AbstractConfigLoader` inherit from `UserDict`,  `OmegaConfigLoader` only presents traits for dictionary access pattern, i.e. dict["key"], but not the other API such as iterating the keys.

## Development notes

The PR is not a perfect fix, but nonetheless address two issues:
1. `__repr__` is misleading as it is missing some arguments to start with, to do this I also have to save `self._custom_resolver`)
2. `keys` is implemented so that at least it is iterable, we can consider `__iter__` and `values` as well but I want to keep this PR minimal.

### Before change
image

### After change
image

## Developer Certificate of Origin
We need all contributions to comply with the [Developer Certificate of Origin (DCO)](https://developercertificate.org/). All commits must be signed off by including a `Signed-off-by` line in the commit message. [See our wiki for guidance](https://github.com/kedro-org/kedro/wiki/Guidelines-for-contributing-developers/).

If your PR is blocked due to unsigned commits, then you must follow the instructions under "Rebase the branch" on the GitHub Checks page for your PR. This will retroactively add the sign-off to all unsigned commits and allow the DCO check to pass.

## Checklist

- [ ] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Signed off each commit with a [Developer Certificate of Origin (DCO)](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/managing-the-commit-signoff-policy-for-your-repository)
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes
- [ ] Checked if this change will affect Kedro-Viz, and if so, communicated that with the Viz team

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -77,6 +79,7 @@ Many thanks to the following Kedroids for contributing PRs to this release:
 * Fixed error handling message for malformed yaml/json files in OmegaConfigLoader.
 * Fixed a bug in `node`-creation allowing self-dependencies when using transcoding, that is datasets named like `name@format`.
 * Improved error message when passing wrong value to node.
+* Fixed a bug when `OmegaConfigLoader` is printed, there is a few arguments missing.
```
Comment (ElenaKhaustova) [inline RELEASE.md:82]: Is it here intentionally?

**Review by ElenaKhaustova — APPROVED** [review]
Approving with a question, the rest LGTM!

File: RELEASE.md (hunk 1 above)
Comment (ankatiyar) [reply RELEASE.md:82]: +1

**Review by ankatiyar — APPROVED** [review]

File: RELEASE.md (hunk 1 above)
Comment (noklam) [reply RELEASE.md]: No idea how this get in😅 nice catch!

**Pull request merged**